### ISSUE TITLE
No '@file' command line option support

### ISSUE THREAD

**Issue by KierenP** [rui314/mold#1124]
I'm trying to replace usage of lld with mold and I've run into the following issue. As seen in the [ld manual](https://linux.die.net/man/1/ld) there exists support for a '@file' command line option. This option is also supported in lld, but not in mold.

```
@file
Read command-line options from file. The options read are inserted in place of the original @file option. If file does not exist, or cannot be read, then the option will be treated literally, and not removed.
Options in file are separated by whitespace. A whitespace character may be included in an option by surrounding the entire option in either single or double quotes. Any character (including a backslash) may be included by prefixing the character to be included with a backslash. The file may itself contain additional @file options; any such options will be processed recursively.
```

Is there any plan to add support for '@file' command line options in the future?

**Comment by rui314** [comment]
The feature is actually supported but is missing in the manual.

**Comment by KierenP** [comment]
I tried a 'hello world' example and you're right. The '@' flag works as expected. 

In my use case, the software I'm building is very large and there are hundreds of '@' arguments and archives being passed to the mold linker. I kept stripping down the example until mold accepted it.

When my command is 65805 characters long it fails with:
```
mold: fatal: cannot open @<path>: No such file or directory
```

When my command is 65660 characters long, mold manages to link the binary.

It seems I'm clearly running into some kind of internal command line limit of 65535 characters for the mold arguments. Is there a way around this?

**Comment by KierenP** [comment]
For a reproducible example, 

main.cpp:
```
#include <iostream>

int main()
{
    std::cout << "Hello, World!" << std::endl;
    return 0;
}
```

object_flags.txt is an empty file

Trying to run the following command with a sufficiently large number of '-Wl,@object_flags.txt' arguments :
```
<path_to_compiler> -fuse-ld=<path_to_mold> -Wl,@object_flags.txt -Wl,@object_flags.txt ... -Wl,@object_flags.txt -Wl,@object_flags.txt main.cpp
```

Results in:
```
mold: fatal: cannot open @object_flags.txt: No such file or directory
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by rui314** [comment]
OK, so the thing is mold cannot read nested response files ('@' files.) I'll fix it.

**Comment by rui314** [comment]
Can you try again with the above commit?

**Comment by KierenP** [comment]
I can confirm it's fixed in ff689a12533a29f029968fb0f890e9bbf914a276 and was broken in ae427d133e3530315a083d86261472abd9e33237. Thanks 👍 

Looking at your change, I don't understand how you fixed the bug I was facing though. I wasn't using nested response files.  I was just doing something like this:

```
<path_to_compiler> -fuse-ld=<path_to_mold> -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt -Wl,@object_flags.txt  main.cpp
```

With about 5000+ empty response files.

**Comment by rui314** [comment]
I believe your compiler automatically creates a response file as a temporary file and pass it to the linker if the command line gets very long. There's a limitation on the maximum length of the command line arguments at the kernel level, so the compiler has to do that.
